README.md

```diff
@@ -1,6 +1,75 @@
 # External Health Monitor
 
-This repo contains sidecar controller and agent for volume health monitoring.
+The External Health Monitor is part of Kubernetes implementation of [Container Storage Interface (CSI)](https://github.com/container-storage-interface/spec). It was introduced as an Alpha feature in Kubernetes v1.19.
+
+## Overview
+
+The [External Health Monitor](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1432-volume-health-monitor) is impelmeted as two components: `External Health Monitor Controller` and `External Health Monitor Agent`.
+
+- External Health Monitor Controller:
+  - The external health monitor controller will be deployed as a sidecar together with the CSI controller driver, similar to how the external-provisioner sidecar is deployed.
+  - Trigger controller RPC to check the health condition of the CSI volumes.
+  - The external controller sidecar will also watch for node failure events. This component can be enabled via a flag.
+
+- External Health Monitor Agent:
+  - The external health monitor agent will be deployed as a sidecar together with the CSI node driver on every Kubernetes worker node.
+  - Trigger node RPC to check volume's mounting conditions.
+  - Note that currently we do not have CSI support for local storage. When the support is available, we will implement relavant CSI monitoring interfaces as well.
```

Comment:
Please remove this sentence: "Note that currently we do not have CSI support for local storage. When the support is available, we will implement relavant CSI monitoring interfaces as well."